src/arithmetic.jl

```diff
@@ -530,14 +592,14 @@ c_k = \sum_{j=0}^k a_j b_{k-j}.
 
 
 """
-    mul!(c, a, b) --> nothing
+    mul!(c, a, b, d=1) --> nothing
 
-Accumulates in `c[k]` the result of `a*b` with minimum allocation; all arguments
-are `HomogeneousPolynomial`.
+Accumulates in `c` the result of `a*b*d` with minimum allocation. Arguments
+c, a and b are `HomogeneousPolynomial`; d is a NumberNotSeries whose default value is 1.
 
 """
 @inline function mul!(c::HomogeneousPolynomial, a::HomogeneousPolynomial,
-        b::HomogeneousPolynomial)
+        b::HomogeneousPolynomial, d::T=1) where {T<:NumberNotSeries}
```

Comment:
I do see the interest to have this method. Yet, I wonder if it does induce some sort of type unstability... My point is, since the default value is `1::Int`, if we set it to something else (say 1.0), do we use the same method or a different one? Perhaps it is better to have two methods, `mul!(c, a, b)` and `mul!(c, a, b, d::T)`, where the latter includes multiplying by the "scalar" `d::T<:NumberNotSeries`.

Reply:
I agree there's some potential for type-instability here; this way it just seemed really convenient to avoid duplicating code. I'll see if I can find another solution.

Reply:
Regarding this point, the current approach is to add `scalar::NumberNotSeries` as a keyword argument, which avoids dispatching on this argument. Adding a `mul!(a, b, c, d::T)` can call the wrong method of `mul!` when `d` is an integer. Another approach can also be to simply define `mul_scalar!(a, b, c, d::NumberNotSeries)` and call this method where appropriate. What are your thoughts on this?

Reply:
I favor your the proposal with `mul_scalar!`, simply because it will have less methods than `mul!`...